gems/pending/openstack/openstack_handle/pagination/base.rb

```diff
@@ -0,0 +1,24 @@
+module OpenstackHandle
+  module Pagination
+    class Base
+      def initialize(service, os_handle, collection_type, options = {}, method = :all)
+        @service         = service
+        @os_handle       = os_handle
+        @collection_type = collection_type
+        @options         = options
+        @method          = method
+      end
+
+      private
+
+      def call_list_method(collection_type, options, method,
+                           paginate_options = {:limit => @service.default_pagination_limit})
```

Comment:
I think you need to do this in the method and ensure you're merging anything coming in as a parameter.  Otherwise, you'll lose the default `:limit` if the caller doesn't send in a limit in their paginate_options.

Should be something like:

```ruby
def call_list_method(..., paginate_options = {})
  paginate_options = {:limit => @service.default_pagination_limit}.merge(paginate_options)
  ...
end
```